From 9fae7e2079f0ca688f414dbc3e6ed168e1cc0fae Mon Sep 17 00:00:00 2001
From: PavelBARABANOV <pavelbarabanov94@gmail.com>
Date: Mon, 19 Oct 2026 05:33:16 +0300
Subject: [PATCH 27/56] stub

(cherry picked from commit 597bed3ca0c80677fcc73d49fd65d6ec378ec267)
---
 src/core/loader/nsp.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/core/loader/nsp.cpp b/src/core/loader/nsp.cpp
index cfc908b4e..7081798d9 100644
--- a/src/core/loader/nsp.cpp
+++ b/src/core/loader/nsp.cpp
@@ -7,6 +7,7 @@
 #include <vector>
 
 #include "common/common_types.h"
+#include "common/string_util.h"
 #include "core/core.h"
 #include "core/file_sys/common_funcs.h"
 #include "core/file_sys/content_archive.h"
@@ -56,6 +57,12 @@ AppLoader_NSP::AppLoader_NSP(FileSys::VirtualFile file_,
 AppLoader_NSP::~AppLoader_NSP() = default;
 
 FileType AppLoader_NSP::IdentifyType(const FileSys::VirtualFile& nsp_file) {
+
+    // .nsz is NSP-compatible; return NSZ so the game list shows the right type.
+    if (name.ends_with(".nsz")) {
+        return FileType::NSZ;
+    }
+
     const FileSys::NSP nsp(nsp_file);
 
     if (nsp.GetStatus() != ResultStatus::Success) {
-- 
2.43.0

